src/PollController.js

```diff
@@ -40,52 +41,77 @@ function (Okta, FormController) {
+      ],
+      _checkStatus: function () {
         // start polling request 
-        this.transaction.poll()
+        this.transactionObject.poll()
           .then((resp) => {
-            this.options.appState.set('transaction', resp);
-            const factorPollingInterval = resp.factor.profile.refresh;
-            let factorPollingIntervalSeconds = Math.floor(factorPollingInterval/1000);
-            // start one second countdown for next poll request
-            this.countDown  = setInterval(() => {
-              // update title after every second and check if countdown == 1 to poll again
-              const title = Okta.loc('polling.title','login', [factorPollingIntervalSeconds]);
-              this.$el.find('.okta-form-title').text(title);
-              if (factorPollingIntervalSeconds === 1) {
-                // restart poll after countdown hits 1
-                this._stopPolling();
-                this._startPolling();
-              } else {
-                // reduce countdown after every second
-                factorPollingIntervalSeconds = factorPollingIntervalSeconds-1;
-              }
-            }, 1000);
+            if (resp.data && resp.status !== 'POLL') {
+              this.options.appState.set('transaction', resp);
+              return;
+            }
+            this.$('.okta-waiting-spinner').hide();
+            let factorPollingIntervalSeconds = Math.ceil(resp.transaction.profile.refresh/1000);
+            this._startCountDown(factorPollingIntervalSeconds);
           })
           .fail(()=> {
-            this._stopPolling();
+            this._stopCountDown();
           });
       },
 
-      _stopPolling: function () {
+      _startCountDown: function (factorPollingIntervalSeconds) {
+        // start one second countdown for next poll request
+        this.countDown  = setInterval(() => {
+          // update title after every second and check if countdown == 1 to poll again
+          this._updateTitle(factorPollingIntervalSeconds);
+          if (factorPollingIntervalSeconds === 0) {
+            /* when countdown hits 0
+                         - stop current poll
+                         - show spinner
+                         - check status
+                        */
+            this._stopCountDown();
+            this.$('.okta-waiting-spinner').show();
+            // start after a small delay so that the spinner does not get hidden too soon
+            setTimeout(() => {
+              this._checkStatus();
+            }, 200);
+          } else {
+            // reduce countdown after every second
+            factorPollingIntervalSeconds = factorPollingIntervalSeconds-1;
+          }
+        }, 1000);
+      },
+
+      _updateTitle: function (factorPollingIntervalSeconds) {
+        this.title = Okta.loc('polling.title','login', [factorPollingIntervalSeconds]);
+        this.$el.find('.okta-form-title').text(this.title);
+      },
+
+      _stopCountDown: function () {
```

Comment:
_stopCountdown should also clear the timeout set on line 85

Reply:
fixed